### PR TITLE
fix:fix the jumping cursor in add friend textfield

### LINKED ISSUE #274: The cursor of the textfield jump to the end when tying at the beginning of the text
When I tried to type `did:key:` in front of the key, the cursor jump to the end while I was typing. 

https://user-images.githubusercontent.com/14248245/200895368-15763cca-acb1-443b-b68c-776a26581f79.mov

I am thinking maybe we don't need `did:key:` when user enter the key, since it become a key icon image in some situation.

### PR DESCRIPTION
**Pull request by hyj1204**
### What this PR does 📖
- Set `value` to `Optional` in the props of IconInput for the textfield that doesn't need a default text. 
The reason why we got that jumping cursor is the previous `value` field keep updating itself through `remote_friend.set(evt.value.clone());`. When code programmatically changes the value of a text field, it resets the cursor position. Since we don't need that default text in this case, so I create a new IconInput without it. 
- Delete the `did:key:` prefix when user copy the did, so user can share their did key directly without 🔑  icon problem.
- Add a closed icon in the IconInput without default text, and by using it to delete all the current text. The reason why I added it is since we don't have `value` field in the input, it lost the ability to delete the text by setting `value` to empty. 

https://user-images.githubusercontent.com/14248245/201138070-02a399a2-7da3-42a4-9562-373f13d72d2a.mov

### Which issue(s) this PR fixes 🔨
- Resolve #274 

### Special notes for reviewers 🗒️
The solution for this issue is still not perfect, because 
~~1) The key code is moved after clicking add button/enter key, so if a user clicks twice without changing any text, the second/later time will show unavailable error(because input is `did:key:`) instead of the error it supposed to.~~ Fixed in PR #294
2) It doesn't delete the text automatically after sending it(clicking add button/enter key). 

The reason why they exist is we delete `value` in input, but if we add it, the jumping cursor problem will come back again. so I will make another issue links to this one. Hope we can find some better solution someday.

### CODE REVIEW

**Review by phillsatellite — APPROVED** [review]

**Comment by phillsatellite** [comment]
Just put the change requested for the Conflicting Files. Functionality of PR was good though and bug was gone 💥

File: src/components/main/friends/sidebar/mod.rs [hunk 1]
```diff
@@ -196,16 +197,15 @@ pub fn FindFriends(cx: Scope, account: Account, add_error: UseState<String>) ->
...
                     placeholder: l.add_placeholder.clone(),
                     icon: Shape::UserAdd,
-                    value: remote_friend.to_string(),
                     on_change: move |evt: FormEvent| {
                         add_error.set(String::new());
                         remote_friend.set(evt.value.clone());
                     },
                     on_enter: move |_| {
-                        let did = DID::try_from(remote_friend.clone().to_string());
+                        let did = DID::try_from(String::from("did:key:") + &(remote_friend.clone().to_string()));
```
Comment (Jekrimo) [inline src/components/main/friends/sidebar/mod.rs]: does a concatenation work here, like 

let did = DID::try_from(`String::from("did:key:") + &(remote_friend.clone().to_string())`)

**Comment by InfamousVague** [comment]
I introduced a small merge conflict, looks good otherwise!

File: src/components/main/friends/sidebar/mod.rs (hunk 1 above)
Comment (Jekrimo) [reply src/components/main/friends/sidebar/mod.rs]: If not, I would say move it out into it's own variable and then pass that variable to try_from

**Review by phillsatellite — APPROVED** [review]

**Review by Jekrimo — APPROVED** [review]

**Review by dariusc93 — APPROVED** [review]
It might be better to check to see if it begins with `did:key:` and then add it if it doesnt but looks good to me from a glance,

**Review by InfamousVague — APPROVED** [review]

File: src/components/main/friends/sidebar/mod.rs (hunk 1 above)
Comment (dariusc93) [reply src/components/main/friends/sidebar/mod.rs]: You could use `format!` macro for this too by doing `format!("did:key:{}", remote_friend.to_string())`

**Pull request merged**